### PR TITLE
👻 Mentee account refactor

### LINKED ISSUE #215: Integrate Mentee Accounts / Refactor Backend
#### Description
Update the API 🌯 so that it supports the new endpoints along with Mentee Accounts as well!

#### Screenshot Todos
image
image

#### Tasks
- [x] Add private field for Mentee Accounts
- [x] Refactor Backend Endpoints/Code
- [x] Integrate into frontend
- [x] Refactor frontend with suggested Kendall tips

### PR DESCRIPTION
**Pull request by Leundai**
## Description :star2:

- Updated backend endpoints to support more than just a single type of account
- Updated frontend API wrapper to support the new endpoint routes
- Created wrapper functions for Mentee/Mentor accounts to avoid changing a lot of code multiple times

Fixes #215 
 
## Screenshots
image

## TODOs :star:
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: frontend/src/utils/api.js [hunk 1]
```diff
@@ -5,42 +5,54 @@ const instance = axios.create({
...
 };
 
-export const uploadMentorImage = (data, id, type) => {
+export const uploadAccountImage = (data, id, type) => {
   let formData = new FormData();
   formData.append("image", data);
-  formData.append("type", type);
-  const requestExtension = "/account/" + id + "/image";
+  formData.append("account_type", type);
+  const requestExtension = `/account/${id}/image`;
```
Comment (kenetec) [inline frontend/src/utils/api.js:55]: using string interpolation * chef's kiss *

File: frontend/src/utils/apiHelper.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import {
```
Comment (kenetec) [inline frontend/src/utils/apiHelper.js:1]: Might be a little weird to have two files for api. I recommend adding all the functions to `api.js` and adding comments to delineate sections (as long as there's no name conflicts).

**Review by kenetec — CHANGES_REQUESTED** [review]
awesome stufff!! I think you just need to move some stuff around and then your PR is golden

**Review by kenetec — APPROVED** [review]
awesome work 🤩

**Review by AngelaLuo49021 — APPROVED** [review]
Love it

**Pull request merged**